examples/nlp/machine_translation/megatron_nmt_training.py

```diff
@@ -83,7 +92,73 @@ def main(cfg) -> None:
     with open_dict(cfg):
         cfg.model.precision = cfg.trainer.precision
 
-    model = MegatronNMTModel(cfg.model, trainer)
+    if hasattr(cfg.model, 'pretrained_model_path') and cfg.model.pretrained_model_path is not None:
+        if not hasattr(cfg.model, 'pretrained_model_type'):
+            raise ValueError(f"Pretrained model type must be in [T5, BART].")
+
+        assert cfg.model.pretrained_model_type in ['T5', 'BART']
+        if cfg.model.pretrained_model_type == 'T5':
+            pretrained_cfg = MegatronT5Model.restore_from(
+                cfg.model.pretrained_model_path, trainer=trainer, return_config=True
+            )
+        else:
+            pretrained_cfg = MegatronBARTModel.restore_from(
+                cfg.model.pretrained_model_path, trainer=trainer, return_config=True
+            )
+        OmegaConf.set_struct(pretrained_cfg, True)
+        with open_dict(pretrained_cfg):
+            pretrained_cfg.masked_softmax_fusion = False
+            # Set source and target language/multilingual
+            pretrained_cfg.src_language = cfg.model.src_language
+            pretrained_cfg.tgt_language = cfg.model.tgt_language
+            pretrained_cfg.multilingual = cfg.model.multilingual
+            pretrained_cfg.shared_tokenizer = True
+
+            # Max generation delta
+            pretrained_cfg.max_generation_delta = cfg.model.max_generation_delta
+
+            # Set label smoothing
+            pretrained_cfg.label_smoothing = cfg.model.label_smoothing
+
+            # Set tokenizer paths:
+            pretrained_cfg.encoder_tokenizer = pretrained_cfg.tokenizer
+            pretrained_cfg.decoder_tokenizer = pretrained_cfg.tokenizer
+
+            # Pre-trained models should use the legacy sentencepiece tokenizer ex: mT5
+            pretrained_cfg.encoder_tokenizer.sentencepiece_legacy = True
+            pretrained_cfg.decoder_tokenizer.sentencepiece_legacy = True
+
+            # Override dropout
+            pretrained_cfg.hidden_dropout = cfg.model.hidden_dropout
```

Comment:
Does it make sense to be able to change the dropout rate when using a pretrained model? Wouldn't the changed dropout cause a change in the distribution and needlessly change the input domain for the upcoming layers?

Reply:
This is useful for cases where you want to regularize your model more while fine-tuning than while training. Check out the mT5 paper where they train with dropout = 0 and fine-tune with dropout 0.1 https://arxiv.org/abs/2010.11934.

Reply:
Hmm... interesting, never paid attention to the changing dropout in the paper. Looks good in this case